pkg/domain/extension.go

```diff
@@ -93,63 +91,50 @@ type ExtensionService struct {
-	// Also used to uniquely identifies an endpoint within the scope of a service
-	URL string
-}
-
-// ExtensionEndpoint is an endpoint through which an extension service can be accessed. Having a list of
+// ExtensionServiceEndpoint is an endpoint through which an extension service can be accessed. Having a list of
 // endpoints associated with a single extension service is particularly important for representing k8s
 // services, which can be exposed both internally (cluster IP) and externally (e.g. ingress). All endpoints
 // belonging to the same extension service must be equivalent in the sense that they are backed by the same
 // API and/or protocol and exhibit the same behavior
-type ExtensionEndpoint struct {
-	// Extension endpoint ID - used to uniquely identify an extension endpoint in the registry
-	ExtensionEndpointID
+type ExtensionServiceEndpoint struct {
+	// Endpoint URL. In case of k8s controllers and operators, the URL points to the cluster API.
+	// Also used to uniquely identifies an endpoint within the scope of a service
+	URL string
 	// Endpoint type - internal/external. An internal endpoint can only be accessed when the consumer
 	// is located in the same zone as the extension service
-	Type ExtensionEndpointType
+	Type ExtensionServiceEndpointType
 	// Configuration entries (e.g. CA certificates), expressed as set of key-value entries
 	Configuration map[string]string
+	// The time when the extension was registered
+	Created time.Time
+	// The time when the extension was last updated
+	Updated time.Time
 }
 
-// ExtensionCredentialsID is a unique extension credentials identifier
-type ExtensionCredentialsID struct {
-	// Extension ID - references the extension this set of credentials belongs to
-	ExtensionID string
-	// Extension service ID - references the extension service this set of credentials belongs to
-	ServiceID string
-	// Extension credentials ID - used to uniquely identify a set of credentials within the scope
-	// of an extension service
-	ID string
-}
-
-// ExtensionCredentials is a group of configuration values that can be generally used to embed information
+// ExtensionServiceCredential is a group of configuration values that can be generally used to embed information
 // pertaining to the authentication and authorization features supported by a service. This descriptor allows
 // administrators and operators of 3rd party tools integrated with FuseML to configure different accounts
 // and credentials (tokens, certificates, passwords etc.) to be associated with different FuseML organization
 // entities (users, projects, groups etc.). All information embedded in a credentials descriptor entry is
 // treated as sensitive information. Each credentials entry has an associated scope that controls who has
 // access to this information (e.g. global, project, user, workflow). This is the equivalent of a k8s secret.
-type ExtensionCredentials struct {
+type ExtensionServiceCredential struct {
```

Comment:
This is (so far) the only thing about this refactoring that I don't agree with: using "Credential" (singular) just to enforce the wider naming scheme. Some English nouns don't play well by these rules. This is one of them. The singular of credentials is never used in the context of authentication/authorization.

Reply:
I have no strong opinion on that, the reason of changing it to `Credential` was to make it more consistent with the other types (including https://github.com/fuseml/fuseml-core/blob/f257ab2395013cae6751f4b5c467df1c5ba20cb3/pkg/domain/extension.go#L21) and use function name conventions. If you think that keeping the plural form is more important I am ok reverting that change.

On the other side, I disagree with your statement. "Credentials" might be more commonly used, however stating that `The singular of credentials is never used in the context of authentication/authorization` is not true at all.